crates/valence_core/src/text/color.rs

```diff
@@ -0,0 +1,368 @@
+//! [`Color`] and related data structures.
+
+use std::fmt;
+use std::hash::Hash;
+
+use serde::de::Visitor;
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+use thiserror::Error;
+
+/// Text color
+#[derive(Default, Debug, PartialOrd, Eq, Ord, Clone, Copy)]
+pub enum Color {
+    /// The default color for the text will be used, which varies by context
+    /// (in some cases, it's white; in others, it's black; in still others, it
+    /// is a shade of gray that isn't normally used on text).
+    #[default]
+    Reset,
+    /// RGB Color
+    Rgb(RgbColor),
+    /// One of the 16 normal Minecraft colors
+    Normal(NormalColor),
+}
+
+/// RGB Color
+#[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash, Debug)]
+pub struct RgbColor {
+    /// Red channel
+    pub r: u8,
+    /// Green channel
+    pub g: u8,
+    /// Blue channel
+    pub b: u8,
+}
+
+/// Normal Minecraft color
+#[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash, Debug)]
+pub enum NormalColor {
```

Comment:
What do you think about the `NamedColor` name instead of `NormalColor`? Is it more descriptive than the current one?